packages/smooth_app/lib/data_models/product_query_model.dart

```diff
@@ -11,60 +8,32 @@ enum LoadingStatus {
 }
 
 class ProductQueryModel with ChangeNotifier {
-  ProductQueryModel(this._supplier, this._daoProduct) {
-    _clear();
+  ProductQueryModel(this._supplier) {
     _asyncLoad(notify: true);
   }
 
   ProductListSupplier _supplier;
-  final DaoProduct _daoProduct;
 
   ProductListSupplier get supplier => _supplier;
 
-  static const String _CATEGORY_ALL = 'all';
-  late String currentCategory;
-
   late LoadingStatus _loadingStatus;
   String? _loadingError;
-  final List<String> _barcodes = <String>[];
-  List<String>? displayBarcodes;
-  bool isNotEmpty() => _barcodes.isNotEmpty;
-  final Map<String, List<String>> _productCategories = <String, List<String>>{};
-
-  /// <Label, Label (count)> [Map]
-  final Map<String, String> categories = <String, String>{};
-
-  /// <Label, count> [Map]
-  final Map<String, int> _categoriesCounter = <String, int>{};
-
-  /// Sorted labels
-  final List<String> sortedCategories = <String>[];
+  List<String> displayBarcodes = <String>[];
+  bool isNotEmpty() => displayBarcodes.isNotEmpty;
 
   String? get loadingError => _loadingError;
   LoadingStatus get loadingStatus => _loadingStatus;
 
-  /// Sets the translation for meta category "All".
-  void setTranslationForAll(final String translationForAll) =>
-      categories[_CATEGORY_ALL] = translationForAll;
-
-  void _clear() {
-    currentCategory = _CATEGORY_ALL;
```

Comment:
Maybe some `///` about the params could be useful

Reply:
@VaiTon Honestly, I would prefer not to.
* we're talking about a _private_ method
* I take care in naming my parameters in a sensible way, and I don't think there's that much ambiguity in `bool notify` or `bool fromScratch` parameters in the context of loading data
* even if there's an ambiguity, the whole file is only 80 lines big, which can give  developers the opportunity to embrace the whole algorithm

That said, I can rename my parameters if needed. What about
* `bool notifyAfterwards`?
* `bool resetDataIfSuccessful`?

Reply:
I would go with `notifyListeners` and the second one is good for me. That being said, it's good for me even in the current state :smile:  !

Reply:
> That being said, it's good for me even in the current state 😄 !

@VaiTon Then I don't change anything, because `notifyListeners` is _also_ the name of the method I call and I would like to avoid the code tap dancing with a parameter and a method that have the same name. Let alone whoever who will have to read it afterwards :)